src/runtime/pkg/containerd-shim-v2/wait.go

```diff
@@ -151,7 +151,7 @@ func watchOOMEvents(ctx context.Context, s *service) {
 				// If the GetOOMEvent call is not implemented, then the agent is most likely an older version,
 				// stop attempting to get OOM events.
 				// for rust agent, the response code is not found
-				if isGRPCErrorCode(codes.NotFound, err) || err.Error() == "Dead agent" {
+				if isGRPCErrorCode(codes.NotFound, err) || err.Error() == "DeadAgent" {
```

Comment:
`err.Error() == "DeadAgent"` will not work, since `kata_agent.go` has changed the error to DeadAgent(xxx):
```golang
 err = fmt.Errorf("DeadAgent(%s)", k.deadErr)
```

How about add a util func `isDeadAgentError`:

```golang
func isDeadAgentError(err error) bool {
    return err != nil && strings.HasPrefix(err.Error(), "DeadAgent")
}
```